pkg/util/application/releases.go

```diff
@@ -21,7 +21,14 @@ func GetContender(appName string, rels []*shipper.Release) (*shipper.Release, er
 // An incumbent release is the first release in this slice that is considered
 // completed.
 func GetIncumbent(appName string, rels []*shipper.Release) (*shipper.Release, error) {
+	contender, err := GetContender(appName, rels)
+	if err != nil {
+		return nil, err
+	}
 	for _, r := range rels {
+		if r == contender {
```

Comment:
This compares two pointers. I don't think the API makes any guarantees about the safety of doing so. Better to compare by name.

Reply:
`GetContender` returns the pointer to the contender if present in `rels`, so this should be safe.

https://github.com/bookingcom/shipper/blob/3c1a7d519962b94fea405ab5df1367c5700e28ce/pkg/util/application/releases.go#L18-L30

Reply:
The API makes no guarantees about the safety of that. It's working today but tomorrow it might return a shallow copy. Either explicitly document the guarantee or compare by name (better).

Reply:
Can you point where this is stated? I'd love to learn more about that, as I thought the assumption is one shouldn't update an object obtained through the API, but not regarding the value themselves.

Reply:
Comparison is based on name equality, not pointers: https://github.com/bookingcom/shipper/pull/166/files#diff-80b2e724ff7817e14b9de5d4d3a25c9cR30